core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -135,11 +139,12 @@ public DefaultServiceRequestContext(
             ServiceConfig cfg, Channel ch, MeterRegistry meterRegistry, SessionProtocol sessionProtocol,
             RequestId id, RoutingContext routingContext, RoutingResult routingResult, HttpRequest req,
             @Nullable SSLSession sslSession, ProxiedAddresses proxiedAddresses, InetAddress clientAddress,
-            long requestStartTimeNanos, long requestStartTimeMicros) {
+            long requestStartTimeNanos, long requestStartTimeMicros, @Nullable EventLoop serviceWorkerGroup) {
```

Comment:
How about just taking `eventLoop` so that the caller specifies the event loop for this context.
For example:
```java
final EventLoop eventLoop;
if (serviceWorkerGroup != null) {
    eventLoop = serviceWokerGroup.next();
} else {
    eventLoop = ch.eventLoop();
}
// Create ServiceRequestContext using the `eventLoop`
...
```

```suggestion
            long requestStartTimeNanos, long requestStartTimeMicros, @Nullable EventLoop eventLoop) {
```